native/java/org/jpype/JPypeSignal.java

```diff
@@ -0,0 +1,47 @@
+package org.jpype;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+
+/**
+ * Java wants to make this action nearly impossible.
```

Comment:
Did you start reading OpenJDK source code? :+1: 
Or hell loads of reverse engineering?

Reply:
It was necessary to read OpenJDK source code to pull this one off.   I started reading to try to solve local reference counting.  But it was do compelling that I had to stick around for the section on signals.  Absolutely riveting, I just can't wait to find out what will happen to little oops next.

Reply:
Thats one of the greatest benefits of open implementations. I hope you will not discover more of these nasty oops any soon...